library/core/src/iter/traits/iterator.rs

```diff
@@ -1761,11 +1761,11 @@ pub trait Iterator {
     /// Basic usage:
     ///
     /// ```
-    /// let a = [1, 2, 3];
+    /// let a = vec![1, 2, 3];
     ///
-    /// let (even, odd): (Vec<i32>, Vec<i32>) = a
-    ///     .iter()
-    ///     .partition(|&n| n % 2 == 0);
+    /// let (even, odd) = a
+    ///     .into_iter()
+    ///     .partition::<Vec<_>, _>(|n| n % 2 == 0);
```

Comment:
Personally, I think it's more readable as:

```suggestion
    /// let (even, odd): (Vec<_>, Vec<_>) = a
    ///     .into_iter()
    ///     .partition(|n| n % 2 == 0);
```

Reply:
Can you apply this change as well?

Reply:
I'm not very sure on this, IDE can show the inferred type on variables which IMO is easier to read than the `(a, b): (TypeA, TypeB)` format.

![image](https://user-images.githubusercontent.com/4814426/156845848-6de9b9c7-3a17-43a6-800b-29503f3f228d.png)

Reply:
Yes, but I think it's easier to read `: (Vec<_>, Vec<_>)` than `partition::<Vec<_>, _>`. The asymmetry between `Vec<_>` for the first argument and just `_` for the second reads less well to me.

Reply:
I do agree `: (Vec<_>, Vec<_>)` is easier to read when there is no syntax inference. But in terms of reading this in an IDE, I would say `partition::<Vec<_>, _>` is better supported. 

rust-analyzer in vscode is not able to show types for `(Vec<_>, Vec<_>)`
![image](https://user-images.githubusercontent.com/4814426/156903196-e46e7b69-0879-44b6-ae14-10056acc35ee.png)

but it is able for `partition::<Vec<_>, _>`:
![image](https://user-images.githubusercontent.com/4814426/156903202-3c9c6545-7006-4216-a64b-99d06eac3bf1.png)

intellij generates a pretty verbose syntax as well:
![image](https://user-images.githubusercontent.com/4814426/156903241-74b68334-db48-4628-a5e4-1bea4ca3ed5d.png)


I guess the question is if we want to have the example to show an IDE friendly use case or not.

Reply:
Given that this code is for example purposes and will be displayed outside of an IDE in most cases, I personally would prefer the `: (Vec<_>, Vec<_>)` syntax. However, I tend to not copy-and-paste example code, whereas I know some people do copy-and-paste, and then modify, example code; so I might be biased.

Reply:
Will keep the change to minimum for now.